### DISCUSSION TITLE
Windows "OS Error 123"

### DISCUSSION THREAD

**Discussion by Regnul** [ReagentX/imessage-exporter#330] [Q&A] [answered]
I've done some digging to try and answer my own question but I am stumped. This error is a basic windows error for "The filename, directory name, or volume label syntax is incorrect". I copied my paths and can confirm they are correct. I do not understand why the program declares it cannot begin when some export has already happened. Its the same limited oldest messages each time. I will attach the last few commands I attempted. I am trying an iOS backup I made in iTunes. I am on Windows 11, Rust 1.80.1 , admin rights in cmd. 

My only idea is that maybe the error is referring to something within the backup which is bad, but then why the unable to start message.

<img width="1061" alt="Screenshot 2024-08-27 235524" src="https://github.com/user-attachments/assets/65439c69-7199-47f3-9260-5bcb7d95e145">

**Comment by ReagentX** [comment]
Do you have a group chat with a custom name that contains one of these characters? 

```
* " / \ < > : | ?
```

How did you install `imessage-exporter`?

**Reply by Regnul** [reply]
I did find and remove a group chat with one of these symbols. Re-running your program on a new export still reports this error. Now some contacts apple suggests include a colon but as that is not user decision I would expect that would be alright?

**Reply by ReagentX** [reply, marked as answer]
This is a filename sanitization bug with `imessage-exporter`;  I fixed it in #331. It will be included in the [Prickly Pear](https://github.com/users/ReagentX/projects/15) release.

As a workaround in the meantime, ensure no contacts or group chat names include characters that are disallowed in Windows file paths. Currently, `imessage-exporter` sanitizes `: \ /`, but the other aforementioned characters will cause problems.

**Reply by Regnul** [reply]
thank you! That seems like a bug that would catch people trying to backup many years of random texts. And to answer the other question, I used cargo to get it.

**Reply by ReagentX** [reply]
Can you share a contact name that has one or more of those characters? It doesn't have to be exact, I am just curious what Apple is generating for you.

**Reply by Regnul** [reply]
it would be something like Maybe : Kyle after getting a message from an account associated with an existing contact or a contact of another contact. 

I also had to remember to purge my recently deleted to fully resolve this error. It works now :)

**Reply by ReagentX** [reply]
Interesting, thanks for the information! Glad you were able to find a workaround. Once #331 is fixed, assuming you kept that original backup, you should be able to export it with all of those messages included.
